pkg/controller/provider/netlify/handler.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 SAP SE or an SAP affiliate company. All rights reserved. h file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *
+ */
+
+package netlify
+
+import (
+	"strings"
+
+	"github.com/netlify/open-api/go/models"
+	"github.com/gardener/controller-manager-library/pkg/logger"
+
+	"github.com/gardener/external-dns-management/pkg/dns"
+	"github.com/gardener/external-dns-management/pkg/dns/provider"
+	"github.com/gardener/external-dns-management/pkg/dns/provider/raw"
+)
+
+type Handler struct {
+	provider.DefaultDNSHandler
+	config provider.DNSHandlerConfig
+	cache  provider.ZoneCache
+	access Access
+}
+
+var _ provider.DNSHandler = &Handler{}
+
+func NewHandler(c *provider.DNSHandlerConfig) (provider.DNSHandler, error) {
+	var err error
+
+	h := &Handler{
+		DefaultDNSHandler: provider.NewDefaultDNSHandler(TYPE_CODE),
+		config:            *c,
+	}
+
+	apiToken, err := c.GetRequiredProperty("NETLIFY_API_TOKEN", "apiToken")
```

Comment:
Is the API_TOKEN the only possible authentication method? If there are others it would be useful to support them also.
Is there http-proxy support in the netlify sdk?